stingray/events.py

```diff
@@ -0,0 +1,212 @@
+            PI channels
+
+        pha : float, numpy.ndarray
+            Photon energies
+
+        """
+        self.time = np.array(times, dtype=np.longdouble)
+        self.energy = np.array(energies)
+        self.ncounts = len(times)
+        self.notes = notes
+        self.dt = dt
+        self.mjdref = mjdref
+        self.gti = gti
+        self.pi = pi
+        self.pha = pha
+
+    @staticmethod
+    def from_fits(fname, **kwargs):
+        from .utils import _assign_value_if_none
+        from .io import load_events_and_gtis
+
+        ret = load_events_and_gtis(fname, **kwargs)
+
+        times = ret.ev_list
+        gti = ret.gtis
+
+        pi = _assign_value_if_none(ret.additional_data, "PI", None)
+        pha = _assign_value_if_none(ret.additional_data, "PHA", None)
+        dt = ret.dt
+        mjdref = ret.mjdref
+
+        return EventList(times, gti=gti, pi=pi, pha=pha, dt=dt, mjdref=mjdref)
+
+    def to_lc(self, bin_time, start_time=None, stop_time=None, center_time=True):
+        """
+        Convert event list to a light curve object.
+
+        Parameters
+        ----------
+        bin_time : float
+            Binning time of the light curve
+
+        Other Parameters
+        ----------------
+        start_time : float
+            Initial time of the light curve
+        stop_time : float
+            Stop time of the light curve
+        center_time: bool
+            If False, time is the start of the bin. Otherwise, the center
+
+        Returns
+        -------
+        lc: `Lightcurve` object
+        """
+
+        times, counts = events_to_lc(self.time, bin_time, start_time,
+            stop_time, center_time)
+
+        return Lightcurve(times, counts)
+
+    def set_times(self, lc):
+        """
+        Assign photon arrival times to event list.
+
+        Parameters
+        ----------
+        lc: `Lightcurve` object
+        """ 
+        times = lc_to_events(lc.time, lc.counts)
+        self.time = EventList(times)
+
+    def set_energies(self, spectrum):
+        """
+        Assign energies to event list.
+
+        Parameters
+        ----------
+        spectrum: 2-d array or list
+            Energies versus corresponding fluxes. The 2-d array or list must
+            have energies across the first dimension and fluxes across the
+            second one.
+        """
+
+        self.energy = assign_energies(self.ncounts, spectrum)
+
+    def read(self, filename, format_='pickle'):
+        """
+        Imports EventList object.
+
+        Parameters
+        ----------
+        filename: str
+            Name of the EventList object to be read.
+
+        format_: str
+            Available options are 'pickle', 'hdf5', 'ascii'
+
+        Returns
+        --------
+        If format_ is 'ascii': astropy.table is returned.
+        If format_ is 'hdf5': dictionary with key-value pairs is returned.
+        If format_ is 'pickle': class object is set.
+        """
```

Comment:
I think I would add `fits` as an option here instead of having a separate method above.